### DISCUSSION TITLE
Navigation entries

### DISCUSSION THREAD

**Discussion by saravanan-rajendran** [MisterPhilip/omnibug#235] [General]
I was trying to get the navigation entries that are printed in your addon. Can you please help me in sharing how do you record the urls that are typed by user, even if it is getting redirect how you are capturing. Please share me an idea how it is done.

**Comment by MisterPhilip** [comment]
The navigation entries are exportable; there is an option when the download modal opens for "Include navigation requests". 

If you mean the code, it is using the [webRequest ](https://developer.chrome.com/docs/extensions/reference/webRequest/)API (see https://github.com/MisterPhilip/omnibug/blob/master/src/eventPage.js#L72)

**Reply by saravanan-rajendran** [reply]
Thanks for your time @MisterPhilip . Yes I was looking on for the script to capture the urls been typed by the user. Is there any way can get it. 
Building like an extension is not possible for me.

**Reply by WhitWaldo** [reply]
@saravanan-rajendran Extensions have a different level of access to browser activity (including to the networking engine) than arbitrary scripts running on a page. Assuming this is what you're trying to do, you might look at the [History API](https://developer.mozilla.org/en-US/docs/Web/API/History_API) and see if it contains what you're looking for.

**Reply by saravanan-rajendran** [reply]
@MisterPhilip @WhitWaldo , Am trying to get the URL's that customers are typing in manually or clicking it from the book marked links. I tried  [History API](https://developer.mozilla.org/en-US/docs/Web/API/History_API). But that doesn't gives me option to get the URL. Any way can we get the users typed URL, links clicked from the book-marked link. It would be a great help for me. Thanks in advance.

**Reply by WhitWaldo** [reply]
@saravanan-rajendran  From what runtime environment are you trying to get this? Are you writing your own extension (e.g. the user has approved its installation to the browser) or is this an arbitrary JS script running on a website?

**Reply by saravanan-rajendran** [reply]
@WhitWaldo just a javascript runs on the page. Not an extension.

**Reply by WhitWaldo** [reply]
@saravanan-rajendran: You won't be able to access that information then. 

Bookmarked links exist in the browser context and aren't exposed on the page (e.g. clicking it is an action that's not visible on the current page's DOM). Either clicking it closes the current page context and goes to another URL or it opens a new tab - and unless it's pointing at another page your script is running on, your script lacks domain context.

Same with the navigation bar - it's not something that exists on the DOM and it's not otherwise exposed to the DOM (and thus available to the JS runtime) via any standardized browser API (only available via extension APIs to select browsers).

Sorry!

**Reply by WhitWaldo** [reply]
This information is only available via Omnibug because it's got access to that information as it's an extension. Because an extension must both be manually installed by the user and the permissions granted it to run explicitly by the user, it's got access to more "sensitive" information like exactly what you're looking to obtain directly from the browser itself independent of the page it's necessarily running on.

**Reply by MisterPhilip** [reply]
As @WhitWaldo mentioned, Omnibug gets additional privileges as a browser extension. Depending on your site, you may have luck using the [PopStateEvent](https://developer.mozilla.org/en-US/docs/Web/API/PopStateEvent), but if you're looking for cross-domain/site tracking then it will not be possible.

I'm closing this since it is offtopic to Omnibug and suggest you try on StackOverflow where others can give you more advice.
